server/storage/database.go

```diff
@@ -36,26 +37,19 @@ func NewMySQLStorage(db *sql.DB) *MySQLStorage {
 
 // UpdateCurrent updates multiple TUF records in a single transaction.
 // Always insert a new row. The unique constraint will ensure there is only ever
-func (db *MySQLStorage) UpdateCurrent(gun, role string, version int, data []byte) error {
-	checkStmt := "SELECT count(*) FROM `tuf_files` WHERE `gun`=? AND `role`=? AND `version`>=?;"
-	insertStmt := "INSERT INTO `tuf_files` (`gun`, `role`, `version`, `data`) VALUES (?,?,?,?) ;"
-
-	// ensure immediately previous version exists
-	row := db.QueryRow(checkStmt, gun, role, version)
-	var exists int
-	err := row.Scan(&exists)
-	if err != nil {
-		return err
-	}
-	if exists != 0 {
-		return &ErrOldVersion{}
-	}
+func (db *MySQLStorage) UpdateCurrent(gun string, update MetaUpdate) error {
+	insertStmt := "INSERT INTO `tuf_files` (`gun`, `role`, `version`, `data`) VALUES (?,?,?,?) WHERE (SELECT count(*) FROM `tuf_files` WHERE `gun`=? AND `role`=? AND `version`>=?) = 0"
```

Comment:
Same commend as above. It prevents old versions that didn't already exist being interleaved.